websubhub-ballerina/content_distribution.bal

```diff
@@ -19,47 +19,22 @@ import ballerina/mime;
 
 isolated function processContentPublish(http:Request request, http:Headers headers, 
                                         map<string> params, HttpToWebsubhubAdaptor adaptor) returns http:Response|error {
-    string? topic = params[HUB_TOPIC];
-    if topic is () {
-        return error Error("Could not find the `hub.topic` parameter");
-    } else {
-        string contentType = request.getContentType();
-        UpdateMessage updateMsg = check createUpdateMessage(contentType, topic, request);
-        Acknowledgement|error updateResult = adaptor.callOnUpdateMethod(updateMsg, headers);
-        http:Response response = new;
-        response.statusCode = http:STATUS_OK;
-        if (updateResult is Acknowledgement) {
-            response.setTextPayload("hub.mode=accepted");
-            response.setHeader("Content-type","application/x-www-form-urlencoded");
-        } else if (updateResult is UpdateMessageError) {
-            var errorDetails = updateResult.detail();
-            updateErrorResponse(response, errorDetails["body"], errorDetails["headers"], updateResult.message());
-        } else {
-            var errorDetails = UPDATE_MESSAGE_ERROR.detail();
-            updateErrorResponse(response, errorDetails["body"], errorDetails["headers"], updateResult.message());
-        }
-        return response;
-    }
+    string topic = check retrieveParameter(params, HUB_TOPIC);
+    string contentType = request.getContentType();
+    UpdateMessage updateMsg = check createUpdateMessage(contentType, topic, request);
+    Acknowledgement|error updateResult = adaptor.callOnUpdateMethod(updateMsg, headers);
```

Comment:
Shouldn't this be an `Error` instead of an `error`?

Reply:
Here we are allowing any generic error to be returned from the remote method, so we can not mark it as `websubhub:Error`. This feature is useful in following scenario.
```ballerina
service /hub on new websubhub:Listener(9090) {
    isolated remote function onRegisterTopic(websubhub:TopicRegistration message, http:Headers headers)
                                returns websubhub:TopicRegistrationSuccess|websubhub:TopicRegistrationError|error {
        check authorize(headers, ["register_topic"]);
        // implement logic
    }

    // other remote methods
}

public isolated function authorize(http:Headers headers, string[] authScopes) returns error? {
    string authHeader = check headers.getHeader(http:AUTH_HEADER);
    jwt:Payload auth = check handler.authenticate(authHeader);
    check handler.authorize(auth, authScopes);
}
```